### PR TITLE
feat(allowlist): Add allowlist.yaml feature

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context 

Add allowlist feature in Python.

### Description

- Allowlist yaml feature
- S3 feature
- DynamoDB feature with the following structure:
image

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: providers/aws/allowlist.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+###########################  ALLOWLIST EXAMPLE  ###########################
+Allowlist:
+  Accounts:
+    "123456789012":
+      Checks:
+        "iam_user_hardware_mfa_enabled":
+          Regions:
+            - "us-east-1"
+          Resources:
+            - "user-1"           # Will ignore user-2 in check iam_user_hardware_mfa_enabled
```
Comment (jfagoagas) [inline providers/aws/allowlist.yaml]:
```suggestion
            - "user-1"           # Will ignore user-1 in check iam_user_hardware_mfa_enabled
```

File: providers/aws/allowlist.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            - "user-2"           # Will ignore user-1 in check iam_user_hardware_mfa_enabled
```
Comment (jfagoagas) [inline providers/aws/allowlist.yaml]:
```suggestion
            - "user-2"           # Will ignore user-2 in check iam_user_hardware_mfa_enabled
```

File: providers/aws/lib/allowlist/allowlist.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                return True
+        # If there is a *, it affects to all accounts
+        if "*" in allowlist["Accounts"]:
+            account = "*"
+            if is_allowlisted_in_check(allowlist, account, check, region, resource):
+                return True
+        return False
+    except Exception as error:
+        logger.critical(f"{error.__class__.__name__} -- {error}")
+        sys.exit()
```
Comment (jfagoagas) [inline providers/aws/lib/allowlist/allowlist.py]:
```suggestion
def is_allowlisted(allowlist, audited_account, check, region, resource):
    try:
        if audited_account in allowlist["Accounts"]:
            if is_allowlisted_in_check(allowlist, audited_account, check, region, resource):
                return True
        # If there is a *, it affects to all accounts
        if "*" in allowlist["Accounts"]:
            audited_account = "*"
            if is_allowlisted_in_check(allowlist, audited_account, check, region, resource):
                return True
        return False
    except Exception as error:
        logger.critical(f"{error.__class__.__name__} -- {error}")
        sys.exit()
```

File: providers/aws/lib/allowlist/allowlist.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    # If there is a *, it affects to all regions
+    if "*" in allowlist["Accounts"][account]["Checks"][check]["Regions"]:
+        for elem in allowlist["Accounts"][account]["Checks"][check]["Resources"]:
+            if re.search(elem, resource):
+                return True
+    # Check if there is the specific region
+    if region in allowlist["Accounts"][account]["Checks"][check]["Regions"]:
+        for elem in allowlist["Accounts"][account]["Checks"][check]["Resources"]:
+            if re.search(elem, resource):
+                return True
```
Comment (jfagoagas) [inline providers/aws/lib/allowlist/allowlist.py]:
```suggestion
def is_allowlisted_in_check(allowlist, audited_account, check, region, resource):
    # If there is a *, it affects to all checks
    if "*" in allowlist["Accounts"][account]["Checks"]:
        check = "*"
        if is_allowlisted_in_region(allowlist, audited_account, check, region, resource):
            return True
    # Check if there is the specific check
    if check in allowlist["Accounts"][account]["Checks"]:
        if is_allowlisted_in_region(allowlist, audited_account, check, region, resource):
            return True
    return False


def is_allowlisted_in_region(allowlist, audited_account, check, region, resource):
    # If there is a *, it affects to all regions
    if "*" in allowlist["Accounts"][audited_account]["Checks"][check]["Regions"]:
        for elem in allowlist["Accounts"][audited_account]["Checks"][check]["Resources"]:
            if re.search(elem, resource):
                return True
    # Check if there is the specific region
    if region in allowlist["Accounts"][audited_account]["Checks"][check]["Regions"]:
        for elem in allowlist["Accounts"][audited_account]["Checks"][check]["Resources"]:
            if re.search(elem, resource):
                return True
```

File: providers/aws/lib/allowlist/allowlist.py (hunk 3 above)
Comment (jfagoagas) [reply providers/aws/lib/allowlist/allowlist.py]: Include the logger line trick

File: prowler [hunk 5]
```diff
@@ -186,6 +188,13 @@ if __name__ == "__main__":
         default=None,
         help="Shodan API key used by check ec2_elastic_ip_shodan.",
     )
+    parser.add_argument(
+        "-w",
+        "--allowlist-file",
+        nargs="?",
+        default=allowlist_yaml,
```
Comment (jfagoagas) [inline prowler]:
```suggestion
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Great job @sergargar !! Please check my comments and then can merge 🥳

File: providers/aws/lib/allowlist/allowlist.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        return False
+    except Exception as error:
+        logger.critical(f"{error.__class__.__name__} -- {error}")
+        sys.exit()
+
+
+def is_allowlisted_in_check(allowlist, audited_account, check, region, resource):
+    # If there is a *, it affects to all checks
+    if "*" in allowlist["Accounts"][account]["Checks"]:
```
Comment (MrCloudSec) [inline providers/aws/lib/allowlist/allowlist.py]:
```suggestion
    if "*" in allowlist["Accounts"][audited_account]["Checks"]:
```

File: providers/aws/lib/allowlist/allowlist.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def is_allowlisted_in_check(allowlist, audited_account, check, region, resource):
+    # If there is a *, it affects to all checks
+    if "*" in allowlist["Accounts"][audited_account]["Checks"]:
+        check = "*"
+        if is_allowlisted_in_region(allowlist, audited_account, check, region, resource):
+            return True
+    # Check if there is the specific check
+    if check in allowlist["Accounts"][account]["Checks"]:
```
Comment (MrCloudSec) [inline providers/aws/lib/allowlist/allowlist.py]:
```suggestion
    if check in allowlist["Accounts"][audited_account]["Checks"]:
```

**Review by jfagoagas — APPROVED** [review]
Good job @sergargar !!

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**